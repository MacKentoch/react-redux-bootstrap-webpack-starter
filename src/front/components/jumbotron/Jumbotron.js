// @flow

// #region imports
import React from 'react';
// #endregion

// #region flow types
type Props = {
  children: React.ReactNode,

  ...any,
};
// #endregion

const Jumbotron = ({ children }: Props) => {
  return <div className="jumbotron">{children}</div>;
};

// #region statics
Jumbotron.displayName = 'Jumbotron';
// #endregion

export default Jumbotron;
