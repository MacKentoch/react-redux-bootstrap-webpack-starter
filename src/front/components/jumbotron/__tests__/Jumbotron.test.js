// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import Jumbotron from '../Jumbotron';
// #endregion

describe('Jumbotron component', () => {
  it('renders as expected', () => {
    const aChild = <p>a child</p>;
    const component = renderer
      .create(
        <div>
          <Jumbotron>{aChild}</Jumbotron>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
