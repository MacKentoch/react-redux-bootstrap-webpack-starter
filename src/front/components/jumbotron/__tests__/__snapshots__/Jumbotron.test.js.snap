// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotron component renders as expected 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <p>
      a child
    </p>
  </div>
</div>
`;
